### PR TITLE
Fix zip steam source

### PR DESCRIPTION
**Pull request by zvin**

### CODE REVIEW

File: lib/source-destination/zip.ts [hunk 1]
```diff
@@ -36,9 +36,24 @@ import { NotCapable } from '../errors';
 import { StreamLimiter } from '../stream-limiter';
 import { streamToBuffer } from '../utils';
 
+export function matchSupportedExtensions(filename: string): boolean {
+	const extension = posix.extname(filename);
```
Comment (telphan) [inline lib/source-destination/zip.ts]: Inconsistent alignment

File: lib/source-destination/single-use-stream-source.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+ * limitations under the License.
+ */
+
+import { SourceDestination } from './source-destination';
+
+import { NotCapable } from '../errors';
+import { StreamLimiter } from '../stream-limiter';
+
+export class SingleUseStreamSource extends SourceDestination {
+	private used = false;
```
Comment (telphan) [inline lib/source-destination/single-use-stream-source.ts]: Inconsistent alignment.

File: lib/source-destination/single-use-stream-source.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+	async createReadStream(
+		emitProgress = false,
+		start = 0,
+		end?: number,
+	): Promise<NodeJS.ReadableStream> {
+		if (this.used) {
+			throw new NotCapable('Single use source stream already used');
+		}
+		if (start !== 0) {
+			throw new NotCapable();
```
Comment (telphan) [inline lib/source-destination/single-use-stream-source.ts]: Would be nice to pass a description here.

File: lib/source-destination/single-use-stream-source.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+import { SourceDestination } from './source-destination';
+
+import { NotCapable } from '../errors';
+import { StreamLimiter } from '../stream-limiter';
+
+export class SingleUseStreamSource extends SourceDestination {
+	private used = false;
+
+	constructor(private stream: NodeJS.ReadableStream) {
```
Comment (telphan) [inline lib/source-destination/single-use-stream-source.ts]: Have you thought about using the types provide by 'stream'? Is a bit more up to date.

File: tests/single-use-stream-source.spec.ts [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+ */
+
+import { using } from 'bluebird';
+import { expect } from 'chai';
+import { createReadStream } from 'fs';
+import 'mocha';
+import { join } from 'path';
+
+import { errors, multiWrite, sourceDestination } from '../lib';
+import { tmpFileDisposer } from '../lib/tmp';
```
Comment (telphan) [inline tests/single-use-stream-source.spec.ts]: Have you thought about using 'tmp-promise'? Pretty well maintained library.

File: tests/single-use-stream-source.spec.ts [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+			);
+		});
+		expect(progressEvents.length).to.be.at.least(2);
+		try {
+			await source.createReadStream();
+			expect(true).to.be.false;
+		} catch (error) {
+			expect(error).to.be.an.instanceof(errors.NotCapable);
+		}
+	});
```
Comment (telphan) [inline tests/single-use-stream-source.spec.ts]: Inconsistent alignment.

File: lib/source-destination/single-use-stream-source.ts (hunk 4 above)
Comment (zvin) [reply lib/source-destination/single-use-stream-source.ts]: Any readable node stream should do, so `ReadableStream` seemed appropriate.

File: lib/source-destination/single-use-stream-source.ts (hunk 3 above)
Comment (zvin) [reply lib/source-destination/single-use-stream-source.ts]: added

**Pull request merged**